orangecontrib/text/widgets/owpreprocess.py

```diff
@@ -1,70 +1,706 @@
+            self.off_label.show()
+            self.title_label.setStyleSheet('color: #B0B0B0;')
+        self.change_signal.emit()
+
+    def restore_data(self, data):
+        # Restores the widget state from the input data.
+        raise NotImplementedError
+
+    def export_data(self):
+        # Export the settings for this module instance.
+        return NotImplementedError
+
+    @staticmethod
+    def get_pp_settings():
+        # Returns the dict representation of this portion of a pre-processor.
+        return NotImplementedError
+
+
+class TokenizerModule(PreprocessorModule):
+    DEFAULT_SETTINGS = {
+        'is_enabled': True,
+        'method': 0,
+    }
+
+    NLTKTokenizer, TwitterTokenizer = 0, 1
+    tokenizer_values = {
+        NLTKTokenizer: False,
+        TwitterTokenizer: True
+    }
+    tokenizer_names = {
+        NLTKTokenizer: 'NLTK tokenizer',
+        TwitterTokenizer: 'Twitter tokenizer',
+    }
+
+    tokenizer_method = 0
+
+    def __init__(self, data):
+        if not data:
+            data = self.DEFAULT_SETTINGS
+        PreprocessorModule.__init__(self, 'Tokenizer', False,
+                                    data.get('is_enabled'))
+
+        self.contents = QWidget()
+        contentArea = QVBoxLayout()
+        contentArea.setContentsMargins(15, 10, 15, 10)
+        self.contents.setLayout(contentArea)
+
+        self.group = QButtonGroup(self, exclusive=True)
+        for method in [self.NLTKTokenizer, self.TwitterTokenizer]:
+            rb = QRadioButton(self, text=self.tokenizer_names[method])
+            if method == self.TwitterTokenizer:
+                # Disable until the Twitter tokenizer is available.
+                rb.setEnabled(False)
+            contentArea.addWidget(rb)
+            self.group.addButton(rb, method)
+        self.group.buttonClicked.connect(self.group_button_clicked)
+
+        self.add_widget(self.contents)
+
+        # Restore the previous state, after starting off the layout.
+        self.restore_data(data)
+
+    def group_button_clicked(self):
+        self.tokenizer_method = self.group.checkedId()
+        self.notify_on_change()
+
+    def restore_data(self, data):
+        self.tokenizer_method = data.get('method')
+        b = self.group.button(self.tokenizer_method)
+        b.setChecked(True)
+
+    def export_data(self):
+        return {
+            'is_enabled': self.enabled,
+            'method': self.tokenizer_method,
+        }
+
+    def get_pp_setting(self):
+        return {
+            'use_twitter_tokenizer': self.tokenizer_values.get(
+                    self.tokenizer_method)
+        }
+
+
+class TransformationModule(PreprocessorModule):
+    DEFAULT_SETTINGS = {
+        'is_enabled': True,
+        'method': 0,
+    }
+
+    PorterStemmer, SnowballStemmer, Lemmatizer = 0, 1, 2
+    transformation_values = {
+        PorterStemmer: PS,
+        SnowballStemmer: SS,
+        Lemmatizer: LM,
+    }
+    transformation_names = {
```

Comment:
This is already defined in `Stemmatizer` class. Please, take the names from there. For example: `self.transformation_values.get(self.transformation_method).name`